test/test_query2.cpp

```diff
@@ -5664,18 +5664,21 @@ TEST(Query_NestedDictionaryNull)
     auto col_any = foo->add_column(type_Mixed, "mixed");
 
     const char* dictOfDictOfNull = R"({ "nestedDict": { "nullValue": null }})";
+    const char* dictOfPrim = R"({ "double": 2.5, "nullValue": null })";
 
     foo->create_object().set_json(col_any, R"("not a dictionary")");
     foo->create_object().set_json(col_any, dictOfDictOfNull);
     foo->create_object().set_json(col_any, dictOfDictOfNull);
     foo->create_object().set_json(col_any, dictOfDictOfNull);
+    foo->create_object().set_json(col_any, dictOfPrim);
 
     CHECK_EQUAL(foo->query("mixed['nestedDict']['nullValue'] == null").count(), 3);
     CHECK_EQUAL(foo->query("mixed.nestedDict.nullValue == null").count(), 3);
     CHECK_EQUAL(foo->query("mixed['nestedDict']['foo'] == null").count(), 3);
     CHECK_EQUAL(foo->query("mixed.nestedDict.foo == null").count(), 3);
     CHECK_EQUAL(foo->query("mixed.nestedDict[*] == null").count(), 3);
     CHECK_EQUAL(foo->query("mixed.nestedDict[*].@type == 'null'").count(), 3);
+    CHECK_EQUAL(foo->query("mixed.double IN {2.5, 3.5}").count(), 1);
```

Comment:
`mixed.double == {2.5, 3.5}` now produces different results than `mixed.double IN {2.5, 3.5}`
So I'm not sure that this is the right decision. Operator `IN` and `==` have been treated as equivalent in many places so I worry that this subtle difference will cause issues down the road. What is the server's behaviour?
My understanding is that the regression is due to the left hand side now being treated as a list, rather than a single property. List vs List comparisons are unique because if neither side has a comparison type defined, then the behaviour is exact element by element matching. Instead of fiddling with the right hand side, could we make the left hand side a single property if there is no list there? (`ValueBase::m_from_list = false` when `mixed.double` is a single value)

Reply:
> Operator `IN` and `==` have been treated as equivalent

But perhaps this has been wrong? My motivation for making this change is the conceptual diffenrence in saying that some things are equal and saying that one thing is included in another thing. I read "A IN B" as A is a subset of B. We can discuss this in a wider audience.

Reply:
Another argument for this change is that the Query::in works in this way.

Reply:
`mixed.double == {2.5, 3.5}` should be handled the same way as `double == {2.5, 3.5}` where `double` is a direct property.

Reply:
Query::in does not support operating on collections, just simple properties.

> mixed.double == {2.5, 3.5} should be handled the same way as double == {2.5, 3.5} where double is a direct property.

I agree. From my perspective, mixed.double is currently being treated as a list when it shouldn't be. I think we should fix this by setting `ctrl.path_only_unary_keys = true` for the case when the path contains only unary links and ends in a simple property. (but I haven't been able to figure out where this goes yet)

Reply:
I drop my case for the time being. The example I gave above actully works. It is only in the list vs. list; I find we could make some improvements.

Reply:
I'm a bit confused. The tests you added here seem to be showing a legitimate problem because they don't pass without your associated changes in driver.cpp.
I think something needs to change in order to make this true: "mixed.double == {2.5, 3.5} should be handled the same way as double == {2.5, 3.5} where double is a direct property."